pkg/predicate/predicate.go

```diff
@@ -200,6 +200,36 @@ func (AnnotationChangedPredicate) Update(e event.UpdateEvent) bool {
 	return !reflect.DeepEqual(e.ObjectNew.GetAnnotations(), e.ObjectOld.GetAnnotations())
 }
 
+// LabelChangedPredicate implements a default update predicate function on label change.
+//
+// This predicate will skip update events that have no change in the object's label.
+// It is intended to be used in conjunction with the GenerationChangedPredicate, as in the following example:
+//
+// Controller.Watch(
+//		&source.Kind{Type: v1.MyCustomKind},
+// 		&handler.EnqueueRequestForObject{},
+//		predicate.Or(predicate.GenerationChangedPredicate{}, predicate.LabelChangedPredicate{}))
+//
+// This will be helpful when object's labels is carrying some extra specification information beyond object's spec,
+// and the controller will be triggered if any valid spec change (not only in spec, but also in labels) happens.
+type LabelChangedPredicate struct {
+	Funcs
```

Comment:
Can you use  `Funcs.UpdateFunc` rather than overwriting it?

Reply:
Sorry, maybe I didn't catch your point, I'm just doing the same way like [ResourceVersionChangedPredicate](https://github.com/kubernetes-sigs/controller-runtime/blob/5564be7542e5d99c77118f2faf3fa9664dc965de/pkg/predicate/predicate.go#L121) and [GenerationChangedPredicate](https://github.com/kubernetes-sigs/controller-runtime/blob/5564be7542e5d99c77118f2faf3fa9664dc965de/pkg/predicate/predicate.go#L155), overriding the `Update` method and inherit the others. 

If we want to assign `Funcs.UpdateFunc`, another constructing function will be introduced for this, and I think maybe it's not easy for using.

Reply:
Ah that's how you came up with it. From a user perspective it is just calling a constructor rather than using a type :)

The reason I think it's slightly better is that it might be confusing to have an embedded `UpdateFunc` described as being called for Update events that does not actually get called for Update events :)

Reply:
:) Got your point, but your suggestion is keeping same as the other predicates or change all the predicates to a new way (use a constructor instead of stating a type), but I think the second way maybe a break change for some "legacy" users.
Or maybe we could just use the **constructing** way for the new `LabelChangedPredicate`?